[GCP] [Billing] Improve documentation (#9931)
elastic/integrations: packages/gcp/_dev/build/docs/billing.md

For detailed usage cost data, set the table pattern to `gcp_billing_export_resource_v1`. Detailed tables include the standard fields and additional fields, such as `effective_price`, enabling a more granular view of expenses.

## Configuration Parameters

### dataset_id

> The `dataset_id` is the ID of your BigQuery dataset where your billing data is stored. This is a unique identifier for the dataset within the project. You can find this ID in your Google Cloud Console under BigQuery.

Reviewer: I think we should add these two bullets somewhere in the doc:

```
- Ensure that you have the necessary permissions to access the dataset.
- The dataset should contain the billing export tables provided by GCP.
```

### table_pattern

> The `table_pattern` defines which tables to pull from within the specified dataset. This can be set to either `gcp_billing_export_v1` for standard usage cost data or `gcp_billing_export_resource_v1` for detailed usage cost data.

Reviewer: Do you think adding `parameter` word makes it better? This is a nitpick. Feel free to decide. If yes, then please add this to other options too. Addition of the keyword `parameter` and changing the tone a little should do.

### cost_type

> The `cost_type` specifies the type of cost data you want to retrieve from your billing data.

Reviewer: ```suggestion
The `cost_type` parameter enables you to filter the cost data based on specific cost categories. You can select one of the following options:
```

Reviewer: Just checking. We do not allow multiple values here, right?

- `regular`: This cost type includes all the regular costs associated with your usage of GCP services. This does not include any taxes, adjustments, or rounding errors.

> - `tax`: This cost type includes all the taxes associated with your usage of GCP services. This does not include the regular costs, adjustments, or rounding errors.

Reviewer: ```suggestion
- `tax`: This cost type includes all the taxes associated with your usage of GCP services. This includes all the applicable taxes based on your billing location and the services used.
```

- `adjustment`: This cost type includes any adjustments made to your billing data. Adjustments could be due to a variety of reasons such as credits, discounts, or any other modifications made to the original cost.

- `rounding_error`: This cost type includes any rounding errors that occurred when calculating your costs. These are typically very small amounts and are used to reconcile any discrepancies due to rounding.

## Example Configuration

Here's an example of what your configuration might look like:

```
dataset_id: "my_billing_dataset"
table_pattern: "gcp_billing_export_resource_v1"
project_id: "my_project"
cost_type: "regular"
```

> In this example, the agent will pull data from the `gcp_billing_export_resource_v1` table within the `my_billing_dataset` dataset.

Reviewer: ```suggestion
In this example, the Agent will extract data from the `gcp_billing_export_resource_v1` table in the `my_billing_dataset` dataset.
```

For detailed usage cost data, set the table pattern to `gcp_billing_export_resource_v1`. Detailed tables include the standard fields and additional fields, such as `effective_price`, enabling a more granular view of expenses.

## Requirements

You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.

Before using any GCP integration you will need:

* **GCP Credentials** to connect with your GCP account.
* **GCP Permissions** to make sure the service account you're using to connect has permission to share the relevant data.

> To collect GCP Billing metrics, you would need specific permissions to access the necessary data. Here's a list of permissions required:

Reviewer 2: ```suggestion
To collect GCP Billing metrics, the following permissions are required to access the necessary data:
```
